From 5574e6ed2f055658905a1b5c63331ea07c3ba539 Mon Sep 17 00:00:00 2001
From: jtheertha <jtheertha@75f.io>
Date: Thu, 16 Jun 2022 07:47:21 +0000
Subject: [PATCH 19/50] Merged PR 4091: Dis name for analog input changed for
 piloop and sense.

Dis name for analog input changed for piloop and sense.
Migration task is written for backward compatibility

Related work items: #11871
---
 .../main/java/a75f/io/api/haystack/Queries.kt |   3 +
 .../main/java/a75f/io/api/haystack/Tags.java  |   2 +
 .../hyperstatsense/HyperStatSenseUtil.java    |  26 ++---
 .../io/logic/bo/building/plc/PlcEquip.java    |   2 +-
 .../building/plc/PlcRelayConfigHandler.java   |  28 ++---
 .../point/PointMigrationHandler.java          | 108 ++++++++++++++++++
 .../a75f/io/logic/util/MigrationUtil.java     |  31 ++++-
 .../a75f/io/logic/util/PreferenceUtil.java    |  12 ++
 8 files changed, 178 insertions(+), 34 deletions(-)
 create mode 100644 logic/src/main/java/a75f/io/logic/migration/point/PointMigrationHandler.java

diff --git a/haystack/src/main/java/a75f/io/api/haystack/Queries.kt b/haystack/src/main/java/a75f/io/api/haystack/Queries.kt
index eee1680a4..7721ff31f 100644
--- a/haystack/src/main/java/a75f/io/api/haystack/Queries.kt
+++ b/haystack/src/main/java/a75f/io/api/haystack/Queries.kt
@@ -13,6 +13,9 @@ class Queries {
         const val ANALOG1_IN = "analog1 and in"
         const val ANALOG2_IN = "analog2 and in"
         const val EQUIP_AND_TUNER = "equip and tuner"
+        const val SENSE_EQUIP = "equip and sense"
+        const val PI_LOOP_EQUIP = "equip and pid"
+
 
     }
 
diff --git a/haystack/src/main/java/a75f/io/api/haystack/Tags.java b/haystack/src/main/java/a75f/io/api/haystack/Tags.java
index 675ab5b0c..abb6c27c0 100644
--- a/haystack/src/main/java/a75f/io/api/haystack/Tags.java
+++ b/haystack/src/main/java/a75f/io/api/haystack/Tags.java
@@ -22,6 +22,8 @@ public class Tags
     public static final String CONNECTION = "connection";
     public static final String ID = "id";
     public static final String DIS = "dis";
+    public static final String SHORTDIS= "shortDis";
+    public static final String GROUP= "group";
     public static final String MARKER = "marker";
     public static final String CCU = "ccu";
     
diff --git a/logic/src/main/java/a75f/io/logic/bo/building/hyperstatsense/HyperStatSenseUtil.java b/logic/src/main/java/a75f/io/logic/bo/building/hyperstatsense/HyperStatSenseUtil.java
index c4f22676c..2ca67c525 100644
--- a/logic/src/main/java/a75f/io/logic/bo/building/hyperstatsense/HyperStatSenseUtil.java
+++ b/logic/src/main/java/a75f/io/logic/bo/building/hyperstatsense/HyperStatSenseUtil.java
@@ -236,7 +236,7 @@ public class HyperStatSenseUtil {
 
     public static Bundle getAnalogBundle(int analog) {
         Bundle bundle = new Bundle();
-        String shortDis = "Generic 0-10 Voltage";
+        String shortDis = "Generic 0:10 Voltage";
         String shortDisTarget = "Dynamic Target Voltage";
         String unit = "V";
         String maxVal = "10";
@@ -245,7 +245,7 @@ public class HyperStatSenseUtil {
         String[] markers = null;
         switch (analog) {
             case 0:
-                shortDis = "Generic (0-10)V";
+                shortDis = "Generic (0:10)V";
                 shortDisTarget = "Dynamic Target Voltage";
                 unit = "V";
                 maxVal = "10";
@@ -254,7 +254,7 @@ public class HyperStatSenseUtil {
                 markers = null;
                 break;
             case 1:
-                shortDis = "Pressure Sensor (0-2)inH₂O";
+                shortDis = "Pressure Sensor (0:2)inH₂O";
                 shortDisTarget = "Dynamic Target Pressure";
                 unit = Consts.PRESSURE_UNIT;
                 maxVal = "2";
@@ -263,7 +263,7 @@ public class HyperStatSenseUtil {
                 markers = new String[]{"pressure"};
                 break;
             case 2:
-                shortDis = "Differential Pressure Sensor (0-0.25)inH₂O";
+                shortDis = "Differential Pressure Sensor (0:0.25)inH₂O";
                 shortDisTarget = "Dynamic Target Pressure Differential";
                 unit = Consts.PRESSURE_UNIT;
                 maxVal = "0.25";
@@ -272,7 +272,7 @@ public class HyperStatSenseUtil {
                 markers = new String[]{"pressure"};
                 break;
             case 3:
-                shortDis = "Airflow Sensor (0-1000)cfm";
+                shortDis = "Airflow Sensor (0:1000)cfm";
                 shortDisTarget = "Dynamic Target Airflow";
                 unit = "cfm";
                 maxVal = "1000";
@@ -281,7 +281,7 @@ public class HyperStatSenseUtil {
                 markers = new String[]{"airflow"};
                 break;
             case 4:
-                shortDis = "Humidity (0-100)%";
+                shortDis = "Humidity (0:100)%";
                 shortDisTarget = "Dynamic Target Humidity";
                 unit = "%";
                 maxVal = "100";
@@ -290,7 +290,7 @@ public class HyperStatSenseUtil {
                 markers = new String[]{"humidity"};
                 break;
             case 5:
-                shortDis = "CO2 (0-2000)ppm";
+                shortDis = "CO2 (0:2000)ppm";
                 shortDisTarget = "Dynamic Target CO2 Level";
                 unit = "ppm";
                 maxVal = "2000";
@@ -299,7 +299,7 @@ public class HyperStatSenseUtil {
                 markers = new String[]{"co2"};
                 break;
             case 6:
-                shortDis = "CO (0-100)ppm";
+                shortDis = "CO (0:100)ppm";
                 shortDisTarget = "Dynamic Target CO Level";
                 unit = "ppm";
                 maxVal = "100";
@@ -308,7 +308,7 @@ public class HyperStatSenseUtil {
                 markers = new String[]{"co"};
                 break;
             case 7:
-                shortDis = "NO2 (0-5)ppm";
+                shortDis = "NO2 (0:5)ppm";
                 shortDisTarget = "Dynamic Target NO2 Level";
                 unit = "ppm";
                 maxVal = "5";
@@ -317,7 +317,7 @@ public class HyperStatSenseUtil {
                 markers = new String[]{"no2"};
                 break;
             case 8:
-                shortDis = "CT (0-10)amps";
+                shortDis = "CT (0:10)amps";
                 shortDisTarget = "Dynamic Target Current Draw";
                 unit = "amps";
                 maxVal = "10";
@@ -326,7 +326,7 @@ public class HyperStatSenseUtil {
                 markers = new String[]{"current", "transformer"};
                 break;
             case 9:
-                shortDis = "CT (0-20)amps";
+                shortDis = "CT (0:20)amps";
                 shortDisTarget = "Dynamic Target Current Draw";
                 unit = "amps";
                 maxVal = "20";
@@ -335,7 +335,7 @@ public class HyperStatSenseUtil {
                 markers = new String[]{"current", "transformer"};
                 break;
             case 10:
-                shortDis = "CT (0-50)amps";
+                shortDis = "CT (0:50)amps";
                 shortDisTarget = "Dynamic Target Current Draw";
                 unit = "amps";
                 maxVal = "50";
@@ -344,7 +344,7 @@ public class HyperStatSenseUtil {
                 markers = new String[]{"current", "transformer"};
                 break;
             case 11:
-                shortDis = "ION Meter (0-1 Million)ions/cc";
+                shortDis = "ION Meter (0:1 Million)ions/cc";
                 shortDisTarget = "Dynamic Target ION Density";
                 unit = "ions/cc";
                 maxVal = "10";
diff --git a/logic/src/main/java/a75f/io/logic/bo/building/plc/PlcEquip.java b/logic/src/main/java/a75f/io/logic/bo/building/plc/PlcEquip.java
index b25b3a540..4235eb5a8 100644
--- a/logic/src/main/java/a75f/io/logic/bo/building/plc/PlcEquip.java
+++ b/logic/src/main/java/a75f/io/logic/bo/building/plc/PlcEquip.java
@@ -727,7 +727,7 @@ public class PlcEquip {
         String[] markers = bundle.getStringArray("markers");
 
         Point.Builder dynamicTargetValueTag = new Point.Builder()
-                .setDisplayName(equipDis + "-dynamicTargetValue-" + dynamicTargetTag)
+                .setDisplayName(equipDis + "-dynamicTargetValue-" + dynamicTargetTag.replace("-" , ":"))
                 .setEquipRef(equipRef)
                 .setSiteRef(siteRef)
                 .setRoomRef(roomRef)
diff --git a/logic/src/main/java/a75f/io/logic/bo/building/plc/PlcRelayConfigHandler.java b/logic/src/main/java/a75f/io/logic/bo/building/plc/PlcRelayConfigHandler.java
index 662608e89..74e4eb1b6 100644
--- a/logic/src/main/java/a75f/io/logic/bo/building/plc/PlcRelayConfigHandler.java
+++ b/logic/src/main/java/a75f/io/logic/bo/building/plc/PlcRelayConfigHandler.java
@@ -712,7 +712,7 @@ public class PlcRelayConfigHandler {
 
     public static Bundle getAnalog1Bundle(int analog1InputSensor) {
         Bundle mBundle = new Bundle();
-        String shortDis = "Generic 0-10 Voltage";
+        String shortDis = "Generic 0:10 Voltage";
         String shortDisTarget = "Target Voltage";
         String unit = "V";
         String maxVal = "10";
@@ -722,7 +722,7 @@ public class PlcRelayConfigHandler {
         switch (analog1InputSensor) {
             case 0:
             case 1:
-                shortDis = "Generic 0-10 Voltage";
+                shortDis = "Generic 0:10 Voltage";
                 shortDisTarget = "Target Voltage";
                 unit = "V";
                 maxVal = "10";
@@ -731,7 +731,7 @@ public class PlcRelayConfigHandler {
                 markers = null;
                 break;
             case 2:
-                shortDis = "Pressure [0-2 in.]";
+                shortDis = "Pressure [0:2 in.]";
                 shortDisTarget = "Target Pressure";
                 unit = Consts.PRESSURE_UNIT;
                 maxVal = "2";
@@ -740,7 +740,7 @@ public class PlcRelayConfigHandler {
                 markers = new String[]{"pressure"};
                 break;
             case 3:
-                shortDis = "Pressure[0-0.25 in. Differential]";
+                shortDis = "Pressure[0:0.25 in. Differential]";
                 shortDisTarget = "Target Pressure Differential";
                 unit = Consts.PRESSURE_UNIT;
                 maxVal = "0.25";
@@ -794,7 +794,7 @@ public class PlcRelayConfigHandler {
                 markers = new String[]{"no2"};
                 break;
             case 9:
-                shortDis = "Current Drawn[CT 0-10]";
+                shortDis = "Current Drawn[CT 0:10]";
                 shortDisTarget = "Target Current Draw";
                 unit = "amps";
                 maxVal = "10";
@@ -803,7 +803,7 @@ public class PlcRelayConfigHandler {
                 markers = new String[]{"current", "transformer"};
                 break;
             case 10:
-                shortDis = "Current Drawn[CT 0-20]";
+                shortDis = "Current Drawn[CT 0:20]";
                 shortDisTarget = "Target Current Draw";
                 unit = "amps";
                 maxVal = "20";
@@ -812,7 +812,7 @@ public class PlcRelayConfigHandler {
                 markers = new String[]{"current", "transformer"};
                 break;
             case 11:
-                shortDis = "Current Drawn[CT 0-50]";
+                shortDis = "Current Drawn[CT 0:50]";
                 shortDisTarget = "Target Current Draw";
                 unit = "amps";
                 maxVal = "50";
@@ -931,7 +931,7 @@ public class PlcRelayConfigHandler {
 
     public static Bundle getAnalog2Bundle(int dynamicInputSensor) {
         Bundle mBundle = new Bundle();
-        String shortDis = "Generic 0-10 Voltage";
+        String shortDis = "Generic 0:10 Voltage";
         String shortDisTarget = "Dynamic Target Voltage";
         String unit = "V";
         String maxVal = "10";
@@ -940,7 +940,7 @@ public class PlcRelayConfigHandler {
         String[] markers = null;
         switch (dynamicInputSensor) {
             case 0:
-                shortDis = "Generic 0-10 Voltage";
+                shortDis = "Generic 0:10 Voltage";
                 shortDisTarget = "Dynamic Target Voltage";
                 unit = "V";
                 maxVal = "10";
@@ -949,7 +949,7 @@ public class PlcRelayConfigHandler {
                 markers = null;
                 break;
             case 1:
-                shortDis = "Pressure [0-2 in.]";
+                shortDis = "Pressure [0:2 in.]";
                 shortDisTarget = "Dynamic Target Pressure";
                 unit = Consts.PRESSURE_UNIT;
                 maxVal = "2";
@@ -958,7 +958,7 @@ public class PlcRelayConfigHandler {
                 markers = new String[]{"pressure"};
                 break;
             case 2:
-                shortDis = "Pressure[0-0.25 in. Differential]";
+                shortDis = "Pressure[0:0.25 in. Differential]";
                 shortDisTarget = "Dynamic Target Pressure Differential";
                 unit = Consts.PRESSURE_UNIT;
                 maxVal = "0.25";
@@ -1012,7 +1012,7 @@ public class PlcRelayConfigHandler {
                 markers = new String[]{"no2"};
                 break;
             case 8:
-                shortDis = "Current Drawn[CT 0-10]";
+                shortDis = "Current Drawn[CT 0:10]";
                 shortDisTarget = "Dynamic Target Current Draw";
                 unit = "amps";
                 maxVal = "10";
@@ -1021,7 +1021,7 @@ public class PlcRelayConfigHandler {
                 markers = new String[]{"current", "transformer"};
                 break;
             case 9:
-                shortDis = "Current Drawn[CT 0-20]";
+                shortDis = "Current Drawn[CT 0:20]";
                 shortDisTarget = "Dynamic Target Current Draw";
                 unit = "amps";
                 maxVal = "20";
@@ -1030,7 +1030,7 @@ public class PlcRelayConfigHandler {
                 markers = new String[]{"current", "transformer"};
                 break;
             case 10:
-                shortDis = "Current Drawn[CT 0-50]";
+                shortDis = "Current Drawn[CT 0:50]";
                 shortDisTarget = "Dynamic Target Current Draw";
                 unit = "amps";
                 maxVal = "50";
diff --git a/logic/src/main/java/a75f/io/logic/migration/point/PointMigrationHandler.java b/logic/src/main/java/a75f/io/logic/migration/point/PointMigrationHandler.java
new file mode 100644
index 000000000..9d5dcad26
--- /dev/null
+++ b/logic/src/main/java/a75f/io/logic/migration/point/PointMigrationHandler.java
@@ -0,0 +1,108 @@
+package a75f.io.logic.migration.point;
+
+import android.os.Bundle;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+import a75f.io.api.haystack.CCUHsApi;
+import a75f.io.api.haystack.Point;
+import a75f.io.api.haystack.Queries;
+import a75f.io.api.haystack.Tags;
+import a75f.io.logic.bo.building.hyperstatsense.HyperStatSenseUtil;
+import a75f.io.logic.bo.building.plc.PlcRelayConfigHandler;
+import a75f.io.logic.bo.building.sensors.Sensor;
+import a75f.io.logic.bo.building.sensors.SensorManager;
+
+public class PointMigrationHandler {
+
+    private PointMigrationHandler(){
+
+    }
+
+    public static void updateSenseAnalogInputUnitPointDisplayName(String analog){
+        HashMap<Object, Object> siteMap = CCUHsApi.getInstance().readEntity(Tags.SITE);
+        String siteDis = siteMap.get(Tags.DIS).toString();
+        ArrayList<HashMap<Object, Object>> senseEquips = CCUHsApi.getInstance().readAllEntities(Queries.SENSE_EQUIP);
+        if(!senseEquips.isEmpty()){
+            for(HashMap<Object, Object> senseEquip : senseEquips){
+
+                String equipId = senseEquip.get(Tags.ID).toString();
+                String equipDis = siteDis + "-SENSE-" + senseEquip.get(Tags.GROUP).toString() + "-";
+
+                HashMap<Object, Object> senseAnalog1InputPoint = CCUHsApi.getInstance().readEntity("sense and " +
+                        analog+" and unit and equipRef == \""+ equipId +"\"");
+                if(!senseAnalog1InputPoint.isEmpty()) {
+                    int configVal = CCUHsApi.getInstance().readDefaultVal("sense and "+analog+" and point and " +
+                            "input and sensor and config" +
+                            " and equipRef == \"" + equipId + "\"").intValue();
+                    Bundle bundle = HyperStatSenseUtil.getAnalogBundle(configVal);
+                    String shortDis = bundle.getString(Tags.SHORTDIS);
+                    Point updatePoint = new Point.Builder().setHashMap(senseAnalog1InputPoint)
+                            .setDisplayName(equipDis + shortDis)
+                            .setShortDis(shortDis)
+                            .build();
+                    CCUHsApi.getInstance().updatePoint(updatePoint, updatePoint.getId());
+                }
+            }
+        }
+    }
+
+    public static void updatePILoopAnalog1InputUnitPointDisplayName(){
+        HashMap<Object, Object> siteMap = CCUHsApi.getInstance().readEntity(Tags.SITE);
+        String siteDis = siteMap.get(Tags.DIS).toString();
+        ArrayList<HashMap<Object, Object>> piLoopEquips = CCUHsApi.getInstance().readAllEntities(Queries.PI_LOOP_EQUIP);
+        if(!piLoopEquips.isEmpty()){
+            for(HashMap<Object, Object> piLoopEquip : piLoopEquips){
+
+                String equipId = piLoopEquip.get(Tags.ID).toString();
+                String equipDis = siteDis + "-PID-" + piLoopEquip.get(Tags.GROUP).toString();
+
+                HashMap<Object, Object> piLoopAnalog1InputPoint = CCUHsApi.getInstance().readEntity("pid and " +
+                        " process and variable and equipRef == \""+ equipId +"\"");
+                if(!piLoopAnalog1InputPoint.isEmpty()) {
+                    int configVal = CCUHsApi.getInstance().readDefaultVal("pid and analog1 and point and " +
+                            "input and sensor and config" +
+                            " and equipRef == \"" + equipId + "\"").intValue();
+                    Bundle bundle = PlcRelayConfigHandler.getAnalog1Bundle(configVal);
+                    String shortDis = bundle.getString(Tags.SHORTDIS);
+                    Point updatePoint = new Point.Builder().setHashMap(piLoopAnalog1InputPoint)
+                            .setDisplayName(equipDis + "-processVariable- " + shortDis)
+                            .setShortDis(shortDis)
+                            .build();
+                    CCUHsApi.getInstance().updatePoint(updatePoint, updatePoint.getId());
+                }
+            }
+        }
+    }
+
+    public static void updatePILoopAnalog2InputUnitPointDisplayName(){
+        ArrayList<HashMap<Object, Object>> piLoopEquips = CCUHsApi.getInstance().readAllEntities(Queries.PI_LOOP_EQUIP);
+        if(!piLoopEquips.isEmpty()){
+            for(HashMap<Object, Object> piLoopEquip : piLoopEquips){
+
+                String equipId = piLoopEquip.get(Tags.ID).toString();
+
+                HashMap<Object, Object>piLoopAnalog2InputPoint = CCUHsApi.getInstance().readEntity("pid and " +
+                        " dynamic and target and equipRef == \""+ equipId +"\"");
+                if(!piLoopAnalog2InputPoint.isEmpty()) {
+                    List<Sensor> sensorList =  SensorManager.getInstance().getExternalSensorList();
+                    for(Sensor sensor : sensorList){
+                        String sensorName = sensor.sensorName;
+                        String dynamicTargetDisName = piLoopAnalog2InputPoint.get("dis").toString();
+                        if(dynamicTargetDisName.contains(sensorName)){
+                            String sensorNameWithColon = sensorName.replace("-", ":");
+                            String modifiedDisplayName = dynamicTargetDisName.replace(sensorName, sensorNameWithColon);
+                            Point updatePoint = new Point.Builder().setHashMap(piLoopAnalog2InputPoint)
+                                    .setDisplayName(modifiedDisplayName)
+                                    .build();
+                            CCUHsApi.getInstance().updatePoint(updatePoint, updatePoint.getId());
+                            break;
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
diff --git a/logic/src/main/java/a75f/io/logic/util/MigrationUtil.java b/logic/src/main/java/a75f/io/logic/util/MigrationUtil.java
index 03b135bee..1fbfb959d 100644
--- a/logic/src/main/java/a75f/io/logic/util/MigrationUtil.java
+++ b/logic/src/main/java/a75f/io/logic/util/MigrationUtil.java
@@ -43,6 +43,7 @@ import static a75f.io.logic.bo.building.definitions.Port.ANALOG_OUT_ONE;
 import static a75f.io.logic.bo.building.definitions.Port.ANALOG_OUT_TWO;
 import a75f.io.logic.diag.DiagEquip;
 import kotlin.Pair;
+import a75f.io.logic.migration.point.PointMigrationHandler;
 
 public class MigrationUtil {
     
@@ -57,7 +58,12 @@ public class MigrationUtil {
             updateAhuRefForBposEquips(CCUHsApi.getInstance());
             PreferenceUtil.setMigrationVersion()
         }*/
-    
+        if(!PreferenceUtil.isSenseAndPILoopAnalogPointDisMigrationDone()){
+            updateAnalogInputDisplayNameForSense();
+            updateAnalogInputDisplayNameForPILOOP();
+            PreferenceUtil.setSenseAndPILoopAnalogPointDisMigrationDone(true);
+        }
+
         if (!PreferenceUtil.isBposAhuRefMigrationDone()) {
             updateAhuRefForBposEquips(CCUHsApi.getInstance());
             PreferenceUtil.setBposAhuRefMigrationStatus(true);
@@ -87,17 +93,16 @@ public class MigrationUtil {
             pressureUnitMigration(CCUHsApi.getInstance());
             PreferenceUtil.setPressureUnitMigrationDone();
         }
-
         if(!PreferenceUtil.isAirflowVolumeUnitMigrationDone()){
             airflowUnitMigration(CCUHsApi.getInstance());
             PreferenceUtil.setAirflowVolumeUnitMigrationDone();
         }
-        
+
         if (!PreferenceUtil.isTrueCFMVAVMigrationDone()) {
             trueCFMVAVMigration(CCUHsApi.getInstance());
             PreferenceUtil.setTrueCFMVAVMigrationDone();
         }
-        
+
         if (!PreferenceUtil.getDamperFeedbackMigration()) {
             doDamperFeedbackMigration(CCUHsApi.getInstance());
             PreferenceUtil.setDamperFeedbackMigration();
@@ -157,6 +162,20 @@ public class MigrationUtil {
       return !ccuHsApi.readEntity(query).isEmpty();
     }
 
+    private static void updateAnalogInputDisplayNameForSense(){
+        if(!CCUHsApi.getInstance().readEntity(Tags.SITE).isEmpty()) {
+            PointMigrationHandler.updateSenseAnalogInputUnitPointDisplayName(Tags.ANALOG1);
+            PointMigrationHandler.updateSenseAnalogInputUnitPointDisplayName(Tags.ANALOG2);
+        }
+    }
+
+    private static void updateAnalogInputDisplayNameForPILOOP(){
+        if(!CCUHsApi.getInstance().readEntity(Tags.SITE).isEmpty()) {
+            PointMigrationHandler.updatePILoopAnalog1InputUnitPointDisplayName();
+            PointMigrationHandler.updatePILoopAnalog2InputUnitPointDisplayName();
+        }
+    }
+
     private static void airflowUnitMigration(CCUHsApi ccuHsApi) {
         ArrayList<HashMap<Object, Object>> airflowPoints = ccuHsApi.readAllEntities("point and airflow and sense and unit");
         String updatedAirflowUnit = "cfm";
@@ -179,7 +198,7 @@ public class MigrationUtil {
             }
         });
     }
-    
+
     private static void trueCFMVAVMigration(CCUHsApi haystack) {
        ArrayList<HashMap<Object, Object>> vavEquips = haystack.readAllEntities("equip and vav and not system");
         HashMap<Object,Object> tuner = CCUHsApi.getInstance().readEntity("equip and tuner");
@@ -188,7 +207,7 @@ public class MigrationUtil {
             doMigrationVav(haystack, vavEquips, tunerEquip);
         }
     }
-    
+
     private static void doMigrationVav(CCUHsApi haystack, ArrayList<HashMap<Object,Object>>vavEquips, Equip tunerEquip) {
         //        creating default tuners for vav
         TrueCFMTuners.createDefaultTrueCfmTuners(haystack, tunerEquip, TunerConstants.VAV_TAG, TunerConstants.VAV_TUNER_GROUP);
diff --git a/logic/src/main/java/a75f/io/logic/util/PreferenceUtil.java b/logic/src/main/java/a75f/io/logic/util/PreferenceUtil.java
index 5d3c3aa8e..17ee85a3b 100644
--- a/logic/src/main/java/a75f/io/logic/util/PreferenceUtil.java
+++ b/logic/src/main/java/a75f/io/logic/util/PreferenceUtil.java
@@ -147,6 +147,18 @@ public class PreferenceUtil {
         editor.apply();
     }
 
+    public static boolean isSenseAndPILoopAnalogPointDisMigrationDone() {
+        SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
+        return sharedPreferences.getBoolean("SenseAndPILoopAnalogPointDisMigrationDone", false);
+    }
+
+    public static void setSenseAndPILoopAnalogPointDisMigrationDone(boolean isMigrated) {
+        SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
+        SharedPreferences.Editor editor = sharedPreferences.edit();
+        editor.putBoolean("SenseAndPILoopAnalogPointDisMigrationDone", isMigrated);
+        editor.apply();
+    }
+
     public static boolean areDuplicateAlertsRemoved() {
         return getBooleanPreference(REMOVED_DUPLICATE_ALERTS);
     }
-- 
2.31.0.windows.1

